move/axelar_gateway/sources/versioned/gateway_v0.move

```diff
@@ -0,0 +1,480 @@
+module axelar_gateway::gateway_v0;
+
+use std::ascii::String;
+
+use sui::clock::Clock;
+use sui::hash;
+use sui::table::{Self, Table};
+use sui::address;
+
+use utils::utils;
+
+use version_control::version_control::VersionControl;
+
+use axelar_gateway::auth::AxelarSigners;
+use axelar_gateway::bytes32::{Self, Bytes32};
+use axelar_gateway::channel::{Self, ApprovedMessage};
+use axelar_gateway::message::{Self, Message};
+use axelar_gateway::message_status::{Self, MessageStatus};
+use axelar_gateway::proof;
+use axelar_gateway::weighted_signers;
+use axelar_gateway::message_ticket::MessageTicket;
+use axelar_gateway::events;
+
+// ------
+// Errors
+// ------
+#[error]
+const EMessageNotApproved: vector<u8> = b"trying to `take_approved_message` for a message that is not approved";
+
+#[error]
+const EZeroMessages: vector<u8> = b"no mesages found";
+
+#[error]
+const ENotLatestSigners: vector<u8> = b"not latest signers";
+
+#[error]
+const ENewerMessage: vector<u8> = b"message ticket created from newer versions cannot be sent here";
+
+// ---------
+// CONSTANTS
+// ---------
+const COMMAND_TYPE_APPROVE_MESSAGES: u8 = 0;
+const COMMAND_TYPE_ROTATE_SIGNERS: u8 = 1;
+
+// -----
+// Types
+// -----
+/// An object holding the state of the Axelar bridge.
+/// The central piece in managing call approval creation and signature verification.
+public struct GatewayV0 has store {
+    operator: address,
+    messages: Table<Bytes32, MessageStatus>,
+    signers: AxelarSigners,
+    version_control: VersionControl,
+}
+
+// -----------------
+// Package Functions
+// -----------------
+/// Init the module by giving a CreatorCap to the sender to allow a full `setup`.
+public(package) fun new(
+    operator: address,
+    messages: Table<Bytes32, MessageStatus>,
+    signers: AxelarSigners,
+    version_control: VersionControl,
+): GatewayV0 {
+    GatewayV0 {
+        operator,
+        messages,
+        signers,
+        version_control,
+    }
+}
+
+public(package) fun version_control(self: &GatewayV0): &VersionControl {
+    &self.version_control
+}
+
+#[syntax(index)]
+public fun borrow(self: &GatewayV0, command_id: Bytes32): &MessageStatus {
```

Comment:
these should be internal functions?
```suggestion
fun borrow(self: &GatewayV0, command_id: Bytes32): &MessageStatus {
```

Reply:
these versioned modules shouldn't have any public functions, otherwise it defeats the point of versioning